### ISSUE TITLE
Authorize plugin should expose `optional` settings

### ISSUE THREAD

**Issue by malekjaroslav** [graphql-nexus/schema#449]
It would be great if `fieldAuthorizePlugin` exposed the `optional` settings. That ways user can decide if `authorize` field is required or not.

```
fieldAuthorizePlugin({
   optional: false
})
```

**Comment by filoscoder** [comment]
I'm just experimenting with `fieldAuthorizePlugin`, and just found this issue.
What kind of issue do you get with this plugin?

I didn't get any complaint saying the `authorize` field is required.

**Comment by awinograd** [comment]
@filoscoder you're correct. Authorize is not required now, but some of us would like to require `authorize`.

However I think the initial issue is insufficient. I think optional should allow for `false | true | 'rootOnly'` where `rootOnly` would require it on root level queries & mutations, but not on subfields

**Comment by Dremora** [comment]
@filoscoder that's exactly the point - it would be nice to make it required. It's easy to forget to add this on a few field, inadvertently exposing it, which would be a security issue.
